PLANET-6918: Implement actions listing page

GP-Dan-Tovbein opened the pull request:
Ref: https://jira.greenpeace.org/browse/PLANET-6918

### Description
New UI applied to the default Actions listing page. We should take action listing pages as a reference with the other listing pages. 
I've also left some comments (for more description) in files changed.

### Testing
Navigate to the [actions page](https://www-dev.greenpeace.org/test-sinope/action/) and compare previous layout against the [unstyled Neptune's instance](https://www-dev.greenpeace.org/test-neptune/action/). Also compare against the one reported by [Nordics](https://www-dev.greenpeace.org/denmark/action/).
Don't forget to `enable|disable` the `Listing page pagination` feature from `Planet 4 > Information architecture`

planet-4 commented:
### Test instance is ready :rocket:

:new_moon: [leda](https://www-dev.greenpeace.org/test-leda) | [admin](https://www-dev.greenpeace.org/test-leda/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-leda/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-leda) | [composer-local.json](https://github.com/greenpeace/planet4-test-leda/blob/main/composer-local.json)

:watch: 2022.10.13 18:46:22

GP-Dan-Tovbein commented:
@comzeradd I think you should approve this [check](https://github.com/greenpeace/planet4-master-theme/pull/1812/checks?check_run_id=8891749971).

comzeradd commented:
> @comzeradd I think you should approve this [check](https://github.com/greenpeace/planet4-master-theme/pull/1812/checks?check_run_id=8891749971).

Yes! Can you rebase first? I think it's needed for frontend-tests to pass.

mleray commented on templates/all-actions.twig:
```diff
@@ -0,0 +1,42 @@
+							data-total="{{ posts.pagination.total }}"
+							data-url="{{ url }}"
+							class="load-more-mt btn btn-small btn-secondary mb-5"
+							data-ga-category="Articles List"
```
Shouldn't this category be changed? 🤔

mleray replied:
Same in `templates/tease-taxonomy-action.twig`

GP-Dan-Tovbein replied:
Not sure about this. Do you mean to change to "Actions List" or to any other category?

mleray replied:
Yeah I think `Actions List` sounds good 👍 Maybe double-check with Molly though 🤔

mleray commented on templates/tease-taxonomy-action.twig:
```diff
@@ -0,0 +1,69 @@
+<li id="result-row-{{ post.ID }}" class="d-flex search-result-list-item">
+	{% include "blocks/image.twig" with {post:post, dummy_thumbnail:dummy_thumbnail} %}
```
I think this can be shortened to `{post, dummy_thumbnail}`

mleray commented on templates/tease-taxonomy-action.twig:
```diff
@@ -0,0 +1,69 @@
+						<a
+							href="{{ tag.link }}"
+							class="search-result-item-tag tag-item tag"
+							data-ga-category="Articles List"
```
Here also I think the category should be changed

mleray commented:
Also, this PR would be merged into `master`, but since then we switched to `main` so that should be changed? I think @comzeradd can do that

GP-Dan-Tovbein commented:
> Also, this PR would be merged into `master`, but since then we switched to `main` so that should be changed? I think @comzeradd can do that

Thanks Maud! I've changed to `main`.

GP-Dan-Tovbein commented on parts/action/query-grid.html:
```diff
@@ -0,0 +1,34 @@
+<!-- wp:query { "query":{ "inherit": true }, "displayLayout": { "type": "flex", "columns":4 } } -->
```
I added these new templates with the main reason to differentiate post and action listing templates.
We'll potentially add the CTA into action templates, but not now since it's an MVP.

GP-Dan-Tovbein commented on parts/action/query-list.html:
```diff
@@ -0,0 +1,37 @@
+<!-- wp:query { "query":{ "inherit": true } } -->
```
https://github.com/greenpeace/planet4-master-theme/pull/1812/files#r997371192

mleray commented on templates/tease-taxonomy-action.twig:
```diff
@@ -0,0 +1,69 @@
+<li id="result-row-{{ post.ID }}" class="d-flex search-result-list-item">
+	{% include "blocks/image.twig" with { post, dummy_thumbnail } %}
```
My bad, this shortcut actually doesn't work in twig templates apparently, the page crashes on my local because of it! Sorryyyyy 😬

GP-Dan-Tovbein commented on assets/src/scss/pages/_listing-page.scss:
```diff
@@ -0,0 +1,153 @@
+    margin-top: $sp-1;
+  }
+
+  // Chrome and Safari
```
It's not 100% supported by Firefox, but from version `103` (currently `106.*`) it's been added as an experimental feature which can be enabled if go to the Firefox `about:config` page, then search and toggle `layout.css.has-selector.enabled`.

More info:
https://developer.mozilla.org/en-US/docs/Web/CSS/:has

GP-Dan-Tovbein commented on assets/src/scss/pages/_listing-page.scss:
```diff
@@ -0,0 +1,153 @@
+  font-size: $font-size-sm;
+  font-family: $roboto;
+
+  // Firefox
```
This piece of code will be removed after the `:has` pseudo class is being fully supported by Firefox

GP-Dan-Tovbein commented on src/Post.php:
```diff
@@ -513,8 +513,7 @@ public function reading_time(): ?int {
 		$time = ( new self( $block->context['postId'] ?? null ) )->reading_time();
-
-		return "<span class='article-list-item-readtime'>$time min read</span>";
+		return "<span class='article-list-item-readtime'>" . ($time ? $time . 'min read' : '') . "</span>";
```
I've modified this logic in cases where `$time` is `null` and don't render unnecessarily `min read`

mleray replied:
Maybe instead of returning the span, if there's no `$time` we could return nothing? Or an empty string?

mleray commented on src/Features/ListingPagePagination.php:
```diff
@@ -28,7 +28,7 @@ protected static function name(): string {
 	 */
 	protected static function description(): string {
-		return __( 'Use the new paginated tag, category, author & post type listing pages.', 'planet4-master-theme-backend' );
+		return __( 'Use the new paginated tag, category, author & post and action type listing pages.', 'planet4-master-theme-backend' );
```
Tiny detail, but I think the `&` here before `post` should be replaced by a comma 🙂

mleray reviewed (commented):
On the [example page](https://www-dev.greenpeace.org/test-sinope/action/), I still see the pagination arrows even though there's only one page, I don't think that's intended?

<img width="192" alt="Screenshot 2022-11-01 at 10 00 28" src="https://user-images.githubusercontent.com/6949075/199197992-3bfef910-8df1-4c77-ac0e-f3af08955d92.png">

GP-Dan-Tovbein commented:
> From the [example page](https://www-dev.greenpeace.org/test-sinope/action/), I have a few comments/questions:
> 
>     * I still see the pagination arrows even though there's only one page, I don't think that's intended? Might not be related to your changes though 🙂

Good catch! Well, I'm sure it's not related to this, probably we should take a look the pagination hook 
> 
> <img alt="Screenshot 2022-11-01 at 10 00 28" width="192" src="https://user-images.githubusercontent.com/6949075/199197992-3bfef910-8df1-4c77-ac0e-f3af08955d92.png">
> 
>     * I think that the author name should be grey like the date, and the separator should be blue (see for example the [story listing page](https://www-dev.greenpeace.org/test-sinope/story/)). Also sometimes the separator between tags is a comma instead of a bullet?
>
> 
> <img alt="Screenshot 2022-11-01 at 10 11 12" width="324" src="https://user-images.githubusercontent.com/6949075/199200060-186c492e-5832-4842-8c90-0299b129a22c.png">

I will ask to the design team, as far as I remember they haven't requested this change. But OFC I will ask !
 
>     * When the test instance uses the grid layout from `Planet 4 > Features > 👷 Listing page grid view` which is WIP, I see that other listing pages (such as the [story one](https://www-dev.greenpeace.org/test-sinope/story/)) still look off, but the action one looks better. Did you fix it for this listing page in particular? 🤔

We're now using the actions listing page as the starting point, that's why I also duplicated the `parts` for actions since doing this we won't have issues with NROs because they aren't use them. In addition to that, the idea is to follow the articles list layout. See this [thread](https://greenpeace-gpi.slack.com/archives/G015K63081W/p1666694594364999) for more details!

>     * Also on the [story listing page](https://www-dev.greenpeace.org/test-sinope/story/), I notice some separators appearing when they are not needed, which doesn't seem to be an issue [on production](https://www.greenpeace.org/international/story/) so maybe it's related to your changes?

Hmm, I haven't touched those template through this PR. That template is based on [this template](https://github.com/greenpeace/planet4-master-theme/blob/e26934565c46b331bb8ea5900d369a323ebf8c4d/parts/query-grid.html)  
> 
> <img alt="Screenshot 2022-11-01 at 10 23 52" width="982" src="https://user-images.githubusercontent.com/6949075/199202189-8a7829eb-7556-4d9c-a095-65902b325c86.png">

mleray commented:
> Hmm, I haven't touched those template through this PR. That template is based on [this template](https://github.com/greenpeace/planet4-master-theme/blob/e26934565c46b331bb8ea5900d369a323ebf8c4d/parts/query-grid.html)

Ugh, yeah my bad, sorry it gets confusing with all the templates 😛 I created a [ticket](https://jira.greenpeace.org/browse/PLANET-6952) for this issue!

GP-Dan-Tovbein commented on templates/tease-taxonomy-action.twig:
```diff
@@ -0,0 +1,59 @@
+	</div>
+
+	<div id="tease-{{ post.ID }}" class="search-result-item-body tease tease-{{ post.post_type }}">
+		<div class="search-result-tags top-page-tags">
```
For more info: https://greenpeace-gpi.slack.com/archives/G015K63081W/p1667920973170169

mleray reviewed (approved):
Looks great @dantovbein 🎉 Thank you for your infinite patience 🙏
